### DISCUSSION TITLE
动态路由怎么国际化，我配置了为啥不行

### DISCUSSION THREAD

**Discussion by liushunli** [umijs/umi#11707] [Q&A]
config 配置的静态路由，加上title国际化成功
![image](https://github.com/umijs/umi/assets/12014406/47379c63-cd50-4c95-802a-6d10bc660760)

可是manage这块是动态路由菜单，在动态数据那里加了title没用
![image](https://github.com/umijs/umi/assets/12014406/1f1d349f-23b7-4924-b219-679e2a7f03d4)

语言配置文件：
![image](https://github.com/umijs/umi/assets/12014406/98230b54-2e7c-4690-a7a4-f37274dbcdd6)

![image](https://github.com/umijs/umi/assets/12014406/40d7dfa4-6c6f-4b1d-b90f-400cf36a813d)

页面效果：
manage那块动态路由没有国际化
中文
![image](https://github.com/umijs/umi/assets/12014406/ee1b9c09-16cd-438d-8260-7d934b59b98f)

英文
![image](https://github.com/umijs/umi/assets/12014406/2d0347e7-06ee-4e73-b323-47454cad0517)

**Comment by fz6m** [comment]
动态路由的名字需要自己做格式化。

```ts
import { FormattedMessage } from '@umijs/max'

function Title({ id }: { id: string }) {
  return <FormattedMessage id={id} />
}

export const patchClientRoutes = ({ routes }: any) => {
  const someRoute = routes[0].routes[2]
  someRoute.title = someRoute.name = <Title id="route.title" />
}
```
